// @lukso/lsp-smart-contracts/dist/constants.cjs
'use strict';

const lsp0Contracts = require('@lukso/lsp0-contracts');
const lsp1Contracts = require('@lukso/lsp1-contracts');
const lsp1delegateContracts = require('@lukso/lsp1delegate-contracts');
const lsp3Contracts = require('@lukso/lsp3-contracts');
const lsp4Contracts = require('@lukso/lsp4-contracts');
const lsp5Contracts = require('@lukso/lsp5-contracts');
const lsp6Contracts = require('@lukso/lsp6-contracts');
const lsp7Contracts = require('@lukso/lsp7-contracts');
const lsp8Contracts = require('@lukso/lsp8-contracts');
const lsp9Contracts = require('@lukso/lsp9-contracts');
const lsp10Contracts = require('@lukso/lsp10-contracts');
const lsp12Contracts = require('@lukso/lsp12-contracts');
const lsp14Contracts = require('@lukso/lsp14-contracts');
const lsp17contractextensionContracts = require('@lukso/lsp17contractextension-contracts');
const lsp20Contracts = require('@lukso/lsp20-contracts');
const lsp25Contracts = require('@lukso/lsp25-contracts');

const SupportedStandards = {
  LSP3Profile: lsp3Contracts.LSP3SupportedStandard,
  LSP4DigitalAsset: lsp4Contracts.LSP4SupportedStandard,
  LSP9Vault: lsp9Contracts.LSP9SupportedStandard
};
const INTERFACE_IDS = {
  ERC165: "0x01ffc9a7",
  ERC1271: "0x1626ba7e",
  ERC20: "0x36372b07",
  ERC20Metadata: "0xa219a025",
  ERC223: "0x87d43052",
  ERC721: "0x80ac58cd",
  ERC721Metadata: "0x5b5e139f",
  ERC777: "0xe58e113c",
  ERC1155: "0xd9b67a26",
  ERC725X: "0x7545acac",
  ERC725Y: "0x629aa694",
  LSP0ERC725Account: lsp0Contracts.INTERFACE_ID_LSP0ERC725Account,
  LSP1UniversalReceiver: lsp1Contracts.INTERFACE_ID_LSP1,
  LSP1UniversalReceiverDelegate: lsp1delegateContracts.INTERFACE_ID_LSP1DELEGATE,
  LSP6KeyManager: lsp6Contracts.INTERFACE_ID_LSP6KEYMANAGER,
  LSP7DigitalAsset: lsp7Contracts.INTERFACE_ID_LSP7,
  LSP8IdentifiableDigitalAsset: lsp8Contracts.INTERFACE_ID_LSP8,
  LSP9Vault: lsp9Contracts.INTERFACE_ID_LSP9,
  LSP14Ownable2Step: lsp14Contracts.INTERFACE_ID_LSP14,
  LSP17Extendable: lsp17contractextensionContracts.INTERFACE_ID_LSP17Extendable,
  LSP17Extension: lsp17contractextensionContracts.INTERFACE_ID_LSP17Extension,
  LSP20CallVerification: lsp20Contracts.INTERFACE_ID_LSP20CallVerification,
  LSP20CallVerifier: lsp20Contracts.INTERFACE_ID_LSP20CallVerifier,
  LSP11BasicSocialRecovery: "0x049a28f1",
  LSP25ExecuteRelayCall: lsp25Contracts.INTERFACE_ID_LSP25
};
const ERC725YDataKeys = {
  LSP1: { ...lsp1Contracts.LSP1DataKeys },
  LSP3: { ...lsp3Contracts.LSP3DataKeys },
  LSP4: { ...lsp4Contracts.LSP4DataKeys },
  LSP5: { ...lsp5Contracts.LSP5DataKeys },
  LSP6: { ...lsp6Contracts.LSP6DataKeys },
  LSP8: { ...lsp8Contracts.LSP8DataKeys },
  LSP9: { ...lsp9Contracts.LSP9DataKeys },
  LSP10: { ...lsp10Contracts.LSP10DataKeys },
  LSP12: { ...lsp12Contracts.LSP12DataKeys },
  LSP17: { ...lsp17contractextensionContracts.LSP17DataKeys }
};
const LSP1_TYPE_IDS = {
  ...lsp0Contracts.LSP0_TYPE_IDS,
  ...lsp7Contracts.LSP7_TYPE_IDS,
  ...lsp8Contracts.LSP8_TYPE_IDS,
  ...lsp9Contracts.LSP9_TYPE_IDS,
  ...lsp14Contracts.LSP14_TYPE_IDS
};

exports.ERC725YDataKeys = ERC725YDataKeys;
exports.INTERFACE_IDS = INTERFACE_IDS;
exports.LSP1_TYPE_IDS = LSP1_TYPE_IDS;
exports.SupportedStandards = SupportedStandards;
